src/fsharp/TypedTree.fs

```diff
@@ -4054,6 +4059,16 @@ type AnonRecdTypeInfo =
 
     member x.IsLinked = (match x.SortedIds with null -> true | _ -> false)
     
+[<RequireQualifiedAccess>]
+type ErasedUnionInfo =
+    { /// Common ancestor type for all cases in this union, used for ILgen
+      CommonAncestorTy: TType
```

Comment:
Feels like this should be computed, not stored.  That is not stored in TypedTreePickle data (and since you don't want to compute it on load of pickled data, you should just compute it on demand)

Reply:
@Swoorup I'm looking at `UnsortedCaseSourceIndices` as well - it seems it's only used to print types in the original order?   I believe it's probably best not to do this - we don't do it for other structural types, e.g.

```fsharp
> {| Y = 1; X = 1 |};;
val it : {| X: int; Y: int |} = { X = 1; Y = 1 }
```

If so I think maybe we can just remove ErasedUnionInfo

Reply:
> If so I think maybe we can just remove ErasedUnionInfo

Or, leave it in but empty - I guess there's a chance we will add something at a later point

Reply:
(I'll leave it in for now, it does no harm in the prototype)

Reply:
Yes `UnsortedCaseSourceIndices` is only used for print types in order it was defined in. Only way I could test my changes was using fsi, so was easier to eye-debug issues especially for union with lot of cases :)

But I don't have any strong opinions for either or. 

I agree about the `CommonAncestorTy` not needing to be pickled/unpickled. That was something in the back of my mind too.